hand-eye-calibration/full_hand_eye_sample.py

```diff
@@ -0,0 +1,539 @@
+
+Further explanation of this sample is found in our knowledge base:
+Further explaination of this sample is found in our knowledge base:
+https://zivid.atlassian.net/wiki/spaces/ZividKB/pages/103481384/
+"""
+from pathlib import Path
+import time
+import datetime
+
+import cv2
+import numpy as np
+import zivid
+import rtde.rtde as rtde
+import rtde.rtde_config as rtde_config
+
+
+def quat_to_rotm(quat: np.array) -> np.array:
+    """Convert from quaternion to rotation matrix
+
+    Args:
+        quat: Rotations in quaternions
+
+    Returns:
+        Rotations in degrees as 3x3 rotation matrix
+    """
+
+    q_w = quat[0]
+    q_x = quat[1]
+    q_y = quat[2]
+    q_z = quat[3]
+
+    rot_matrix = np.zeros((3, 3))
+    rot_matrix[0, 0] = 1 - 2 * np.power(q_y, 2) - 2 * np.power(q_z, 2)
+    rot_matrix[0, 1] = 2 * q_x * q_y - 2 * q_z * q_w
+    rot_matrix[0, 2] = 2 * q_x * q_z + 2 * q_y * q_w
+    rot_matrix[1, 0] = 2 * q_x * q_y + 2 * q_z * q_w
+    rot_matrix[1, 1] = 1 - 2 * np.power(q_x, 2) - 2 * np.power(q_z, 2)
+    rot_matrix[1, 2] = 2 * q_y * q_z - 2 * q_x * q_w
+    rot_matrix[2, 0] = 2 * q_x * q_z - 2 * q_y * q_w
+    rot_matrix[2, 1] = 2 * q_y * q_z + 2 * q_x * q_w
+    rot_matrix[2, 2] = 1 - 2 * np.power(q_x, 2) - 2 * np.power(q_y, 2)
+
+    return rot_matrix
+
+
+def rotvec_to_quat(rotvec: np.array, angle: float = None) -> np.array:
+    """Convert from rotation vector to quaternions
+
+    Args:
+        rotvec: Rotation vector
+        angle: Angle of rotation
+
+    Returns:
+        Rotation in quaternions
+    """
+
+    if angle is None:
+        angle = np.linalg.norm(rotvec)
+
+    quat = np.zeros((4))
+    quat[0] = np.cos(0.5 * angle)
+    q_r = np.sqrt((1 - np.power(quat[0], 2)) / np.power(np.linalg.norm(rotvec), 2))
+    quat[1] = q_r * rotvec[0]
+    quat[2] = q_r * rotvec[1]
+    quat[3] = q_r * rotvec[2]
+    quat = quat / np.sign(quat[0])
+
+    return quat
+
+
+def _write_robot_state(
+    con: rtde, input_data, finish_capture: bool = False, camera_ready: bool = False
+):
+    """Write to robot I/O registrer
+
+    Args:
+        con: Connection between computer and robot
+        input_data: Input package containing the specific input data registers
+        finish_capture: Boolean value to robot_state that q_r scene capture is finished
+        camera_ready: Boolean value to robot_state that camera is ready to capture images
+    """
+
+    if finish_capture:
+        input_data.input_bit_register_64 = 1
+    else:
+        input_data.input_bit_register_64 = 0
+
+    if camera_ready:
+        input_data.input_bit_register_65 = 1
+    else:
+        input_data.input_bit_register_65 = 0
+
+    con.send(input_data)
+
+
+def _initialize_robot_sync(host: str, port: int):
+    """Set up communication with UR robot
+
+    Args:
+        host: IP address to host
+        port: Port number
+
+    Returns:
+        Connection to robot
+        Input package containing the specific input data registers
```

Comment:
It is a bit confusing when a function returns something called 'input' as it is not clear what is it input to. Maybe robot_input_data

Reply:
fixed